src/components/Modal/index.tsx

```diff
@@ -1,35 +1,52 @@
-import React, {useContext} from "react";
+import React from "react";
 import "./modal.scss"
-import { DialogContext, actionControlVisibility } from "../../context/DialogContext";
 import { MovieModal } from "../ModalBody/MovieModal";
 import { DeleteModal } from "../ModalBody/DeleteModal";
+import { OutsideAlerter } from "../../containers/OutsideAlerter";
+import { useDispatch, useSelector } from "react-redux";
+import { toggleModal } from "../../store/reducers/appReducer";
 
 interface ModalProps {
-    type: "add" | "edit" | "delete"
+    variable: "isAddModalOpened" | "isEditModalOpened" | "isDeleteModalOpened"
 }
 
-export const Modal:React.FunctionComponent<ModalProps> = ({ type }) => {
-    const { state, dispatch } = useContext(DialogContext)
-    const visible = state[type]
-    const onClose = () => dispatch(
-        actionControlVisibility(type, false)
-    )
+export const Modal:React.FunctionComponent<ModalProps> = ({ variable }) => {
+    const dispatch = useDispatch()
+    const visible = useSelector((state: any) => state.appData[variable])
```

Comment:
I would recommend to opt-in for [reselect library](https://www.npmjs.com/package/reselect). Aside from optimizing re-renders it gives you opportunity to define flexible and re-usable selectors which can make your code look DRY'er.